--- a/win32/win32.c
+++ b/win32/win32.c
@@ -4889,6 +4889,7 @@
     return 0;
 }
 
+#if !defined(__MINGW32__) || !defined(HAVE_CLOCK_GETTIME)
 /* License: Ruby's */
 int
 clock_gettime(clockid_t clock_id, struct timespec *sp)
@@ -4928,7 +4929,9 @@
         return -1;
     }
 }
+#endif
 
+#if !defined(__MINGW32__) || !defined(HAVE_CLOCK_GETRES)
 /* License: Ruby's */
 int
 clock_getres(clockid_t clock_id, struct timespec *sp)
@@ -4956,6 +4959,7 @@
         return -1;
     }
 }
+#endif
 
 /* License: Ruby's */
 static char *
